tools/pymanifold.cpp

```diff
@@ -0,0 +1,169 @@
+#include "manifold.h"
+#include "meshIO.h"
+#include "pybind11/functional.h"
+#include "pybind11/numpy.h"
+#include "pybind11/operators.h"
+#include "pybind11/pybind11.h"
+#include "pybind11/stl.h"
+
+namespace py = pybind11;
+
+using namespace manifold;
+
+void exportManifold(Manifold &m, std::string name) {
+  Mesh out = m.GetMesh();
+  manifold::ExportOptions options;
+  options.faceted = true;
+  options.mat.roughness = 0.2;
+  options.mat.metalness = 0.0;
+  manifold::ExportMesh(name, out, options);
+};
+
+typedef std::tuple<float, float> Float2;
+typedef std::tuple<float, float, float> Float3;
+
+struct PolygonsWrapper {
+  std::unique_ptr<Polygons> polygons;
+};
+
+PYBIND11_MODULE(pymanifold, m) {
+  m.doc() = "Python binding for the manifold library. Please check the C++ documentation for APIs.\n"
+      "This binding will perform copying to make the API more familiar to OpenSCAD users.";
```

Comment:
Can we talk about this in a little more depth? As an OpenSCAD user, I'm not sure this applies, since the OpenSCAD language is so different (yes it's functional, but it's also different in lots of other ways). Do you have a sense of how performance-impacting this decision is? Are there other considerations impacting this choice? 

It would be great to have a cross-compiler that took OpenSCAD language and converted it to an equivalent python or JS program using Manifold, but that seems like the right layer at which to account for the functional nature of OpenSCAD's language, rather than our bindings themselves. What do you think?

Reply:
Regarding performance, I did some simple tests with a few examples and they are all faster (~5x faster on my machihne) than OpenSCAD with fast-csg enabled, at least according to the figures in https://gist.github.com/ochafik/2db96400e3c1f73558fcede990b8a355. You can try running those examples and check the performance.

For more complicated models, I think we should do something like lazy union, which tries to compose most of the manifolds that do not intersect each other. From my benchmark with the [bricks](https://gist.github.com/deckar01/ef11def51de7e71d9f288c6e5819fdb7) example, lazy union is a lot faster for larger models (took about 8s for width=height=length=100). The lazy union feature will also share the underlying list of manifolds, so objects that only differ in transformation will be lazily cloned. When that feature is added, this copying will basically be free.

If we don't do copying, users will have to manually copy the manifolds via constructor/copy function call, which may not be very ergonomic to use. But I agree, maybe I should split it into a low-level binding and a user facing module that provides this kind of features.

Reply:
@rsaccon Since you seem to another interested Python user, would you care to weigh in here, regarding ergonomics and performance?

Reply:
Another consideration for this: If copying causes performance problem, it should not be too hard for us to modify the c++ code to add more laziness to the module using some sort of wrapper, even if lazy union is not yet implemented. But if users depend on the API that will cause mutation for the manifolds, it would be hard for us to apply optimizations later and changing such behavior will be a breaking change.

Reply:
I have only little experience with Python (I wrote a Freecad addon a while ago), but I have a very specific use case why I need  Python bindings: I wanna give it a try to write a Blender addon which will provide a Manifold based way to create precise shapes. So my wish list is: The bindings should be minimal (full manifold core API, but option to skip assimp). I played with the first lazy union based approach of the bindings (from @pca006132 opt branch), worked great, but the name `ManifoldSet` is or was a bit confusing. For me, performance is very important, and lazy union seems a way to get us there, but to provide some valid feedback I need first to play around a bit more with whatever we have (and best option right now is this WIP PR)

Reply:
Agreed about separating out `meshIO` (the only part with the assimp dependency). I'm not super familiar with Python bindings, but it does seem like having two: one for core Manifold and a separate one for MeshIO would be a good idea. Likewise for WASM. I'll likely use WASM with three.js and I'd prefer to use their import/export capabilities and keep the bundle size down.

Reply:
Yes, I will have a look at how to split it into something like `pymanifold.core` and `pymanifold.assimp` or something like that.